core/src/main/java/com/linecorp/armeria/server/healthcheck/CpuHealthChecker.java

```diff
@@ -0,0 +1,174 @@
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.server.healthcheck;
+
+import static java.util.Objects.requireNonNull;
+
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.management.ManagementFactory;
+import java.lang.management.OperatingSystemMXBean;
+import java.lang.reflect.Method;
+import java.util.List;
+import java.util.function.DoubleSupplier;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableList;
+
+import com.linecorp.armeria.common.annotation.Nullable;
+
+/**
+ * A {@link HealthChecker} that reports as unhealthy when the current
+ * CPU usage or CPU load exceeds threshold. For example:
+ * <pre>{@code
+ * final CpuHealthChecker cpuHealthChecker = HealthChecker.of(10, 10);<br>
+ * final boolean healthy = cpuHealthChecker.isHealthy();
+ * }</pre>
+ */
+// Forked from <a href="https://github.com/micrometer-metrics/micrometer/blob/8339d57bef8689beb8d7a18b429a166f6595f2af/micrometer-core/src/main/java/io/micrometer/core/instrument/binder/system/ProcessorMetrics.java">ProcessorMetrics.java</a> in the micrometer core.
+final class CpuHealthChecker implements HealthChecker {
+    private static final Logger logger = LoggerFactory.getLogger(CpuHealthChecker.class);
+
+    private static final MethodHandles.Lookup LOOKUP = MethodHandles.lookup();
+
+    private static final DoubleSupplier currentSystemCpuUsageSupplier;
+
+    private static final DoubleSupplier currentProcessCpuUsageSupplier;
+
+    @Nullable
+    private static final OperatingSystemMXBean operatingSystemBean;
+
+    private static final Class<?> operatingSystemBeanClass;
+
+    private static final List<String> OPERATING_SYSTEM_BEAN_CLASS_NAMES = ImmutableList.of(
+            "com.ibm.lang.management.OperatingSystemMXBean", // J9
+            "com.sun.management.OperatingSystemMXBean" // HotSpot
+    );
+
+    @Nullable
+    @VisibleForTesting
+    static final MethodHandle systemCpuUsage;
+
+    @Nullable
+    @VisibleForTesting
+    static final MethodHandle processCpuUsage;
+
+    static {
+        operatingSystemBeanClass = requireNonNull(getFirstClassFound(OPERATING_SYSTEM_BEAN_CLASS_NAMES));
+        operatingSystemBean = ManagementFactory.getOperatingSystemMXBean();
+        final MethodHandle getCpuLoad = detectMethod("getCpuLoad");
+        systemCpuUsage = getCpuLoad != null ? getCpuLoad : detectMethod("getSystemCpuLoad");
+        processCpuUsage = detectMethod("getProcessCpuLoad");
```

Comment:
It seems like `Usage` and `Load` are used interchangeably across this class. Can we just use `Load` for consistency? Ditto for the other field, parameter names

```suggestion
        final MethodHandle getCpuLoad = detectMethod("getCpuLoad");
        systemCpuLoad = getCpuLoad != null ? getCpuLoad : detectMethod("getSystemCpuLoad");
        processCpuLoad = detectMethod("getProcessCpuLoad");
```